docs: add data isolation content (#77876)
metabase/metabase: docs/permissions/data-isolation-methods.md

---
title: Choosing a data isolation method
summary: Compare the ways Metabase can restrict data access in multi-tenant setups, including row and column security, impersonation, database routing, and locked parameters.
---
 
# Choosing a data isolation method

> Row and column security, impersonation, and database routing are only available on Metabase [Pro](https://www.metabase.com/product/pro) and [Enterprise](https://www.metabase.com/product/enterprise) plans.

Metabase offers several ways to ensure people only see the data they're supposed to see. The best method depends on how you store your data and where you want to enforce access rules:

> 

Reviewer: I'm thinking maybe we should be slightly more generic and not focus _just_ on multi-tenant setup. We do need a multi-tenant permissions setup guide but 50% of our users are still using Metabase for internal analytics, and this info is just as useful to them, if framed correctly

> - If tenant data lives in the same tables, use [row and column security](#row-and-column-security) to filter data, or [impersonation](#impersonation) to enforce access rules.

Reviewer: ```suggestion
- If all data lives in the same database, use [row and column security](#row-and-column-security) to filter data, or [impersonation](#impersonation) to enforce access rules on the database level.
```

Don't use row and column security when:

- People need to write native SQL queries. Row and column security doesn't apply to the SQL editor.
- You need column-level security on MongoDB or Druid, which only support row-level security.

## Impersonation

[Impersonation](./impersonation.md) lets your database control what each person can see. You define roles and row-level security policies in your database. Metabase connects using the role specified by each person's [user attribute](../people-and-groups/managing.md#adding-a-user-attribute), and your database enforces what that role can see.

![Impersonation](./images/connection-impersonation.png)

Use impersonation when:

- You want to centralize access logic in your database instead of in Metabase.
- People need to write native SQL queries. Since your database enforces the rules, the rules apply in the SQL editor too.

Don't use impersonation when:

- Your database [doesn't support it](./impersonation.md#databases-that-support-impersonation).
- You don't want to set up and maintain roles and policies in your database.
- You rely heavily on caching. You can't share cached results between roles.

## Database routing

> [Database routing](./database-routing.md) sends each person's queries to their own database. You build questions and dashboards once against a primary database, and Metabase routes queries to the right destination database based on each person's [user attribute](../people-and-groups/managing.md#adding-a-user-attribute). Database routing requires that each tenant's data lives in a separate database with an identical schema.

Reviewer: ```suggestion
[Database routing](./database-routing.md) sends each person's queries to their own database. You build questions and dashboards once against a primary database, and Metabase routes queries to the right destination database based on each person's [user attribute](../people-and-groups/managing.md#adding-a-user-attribute). Database routing requires that all destination databases have identical schemas.
```
